packages/gatsby-plugin-gatsby-cloud/src/components/Indicator.js

```diff
@@ -82,7 +121,13 @@ export default function Indicator() {
       buildId !== newBuildInfo?.latestBuild?.id &&
       currentBuild?.buildStatus === `SUCCESS`
     ) {
-      setBuildInfo({ ...newBuildInfo, buildStatus: `SUCCESS` })
+      if (await hasPageDataChanged(buildId)) {
```

Comment:
Have you had a chance to try this on a larger site? I think we might need to poll for page data changes for a bit after the build status moves to success since success doesn't mean the page-data is publicly available on the internet yet (Content Sync has this same problem). I believe RINC will help with this since deploys will be faster, but even in a RINC world we'll probably need to compare the page-data a few times until it changes or we stop comparing.

@DanielSLew with RINC are we incorporating deploy/route-building time into the BUILDING status? Once we get to SUCCESS (with RINC) will the files be publicly available?

Reply:
For context, pre-RINC SUCCESS doesn't mean built files are available to request yet

Reply:
yeah I'm seeing this show up as an issue for large content/index pages

Reply:
In Content Sync once the build status goes to success we poll for updated manifest files for 30 seconds and then timeout. Perhaps we should do the same here but for page-data files

Reply:
Yeah, we dont set a build to finished until it is deployed with RINC